cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/BrokerSetAwareGoal.java

```diff
@@ -123,6 +125,8 @@ public boolean isHardGoal() {
    */
   @Override
   protected void initGoalState(ClusterModel clusterModel, OptimizationOptions optimizationOptions) throws OptimizationFailureException {
+    _isPartitionLevelBrokerSetAware = _balancingConstraint.allowPartitionColoring();
```

Comment:
I would not use word `coloring` since it may be confusing. We have been referring to it as `brokerserAwareness`